lib/log.js

```diff
@@ -178,25 +178,59 @@ Lug.prototype.activityEvent = function (event, request, data) {
     return P.resolve()
   }
 
+  // log a flowEvent for all activityEvents
+  this.flowEvent(event, request)
+
+  var info = {
+    event: event
+  }
+
+  if (request.headers['user-agent']) {
+    info.userAgent = request.headers['user-agent']
+  }
+
+  optionallySetService(info, request)
+
+  Object.keys(data).forEach(function (key) {
+    info[key] = data[key]
+  })
+
+  this.logger.info('activityEvent', info)
+  this.statsd.write(info)
+
+  return P.resolve()
+}
+
+// Log a flow metrics event.
+// These events help understand the user flow.
+
+Lug.prototype.flowEvent = function (event, request) {
   var self = this
 
+  if (! event) {
+    this.error({ op: 'log.flowEvent', missingEvent: true })
+    return P.resolve()
+  }
+
+  if (! request || ! request.headers) {
+    this.error({ op: 'log.flowEvent', event: event, missingHeaders: true })
+    return P.resolve()
+  }
+
   return this.metricsContext.gather({
     event: event,
     userAgent: request.headers['user-agent']
   }, request, event).then(
     function (info) {
+      info.event = event
       optionallySetService(info, request)
 
-      Object.keys(data).forEach(function (key) {
-        info[key] = data[key]
-      })
-
-      self.logger.info('activityEvent', info)
-      self.statsd.write(info)
+      self.logger.info('flowEvent', info)
     }
   )
 }
 
+
 function optionallySetService (data, request) {
   // don't overwrite service if it is specified in metricsContext
```

Comment:
Nit, but this comment is inaccurate now that `optionallSetService` is called in two places. Should just say something like "...if it is already set" instead.